### ISSUE TITLE
Controlnet Erorr when trying to use Noise Inversion

### ISSUE THREAD

**Issue by crazygonuts** [pkuliyi2015/multidiffusion-upscaler-for-automatic1111#208]
I can upscale using tiled diffusion with controlnet just fine unless I check noise inversion, if I do that I get this error

```
5:55:33-379575 ERROR    Exception: ControlNet Error: Failed to detect whether an instance is cond or uncond!          ]
15:55:33-380576 ERROR    Arguments: args=('task(ynymo0wg3bi34yv)', 0, '(sfw) (best quality, masterpiece, absurdres,
                         maximum detail, jaw dropping), full body princess, multicolored dress that looks like a
                         cupcake, (rainbow sprinkles on dress), bonnet hat, (wholesome:1.4), in front of a cafe castle,
                         named Sofia, detailed cloth, finely detailed hair, bubbles, candies with cat faces on dress',
                         '(skin, nsfw, naked) (worst quality, low quality, negative_hand-neg:1.4)', [],
                         <PIL.Image.Image image mode=RGBA size=1536x1280 at 0x16BA2735B70>, None, None, None, None,
                         None, None, 30, 0, 4, 0, 1, False, False, 1, 1, 6, 1.5, 1, 0.75, -1.0, -1.0, 0, 0, 0, False,
                         0, 1280, 1536, 1, 0, 0, 32, 0, '', '', '', [], 0, True, 'Mixture of Diffusers', False, True,
                         1024, 1024, 144, 144, 48, 8, 'R-ESRGAN 4x+ Anime6B', 2, True, 50, 1, 1, 64, False, False,
                         False, False, False, 0.4, 0.4, 0.2, 0.2, '', '', 'Background', 0.2, -1.0, False, 0.4, 0.4,
                         0.2, 0.2, '', '', 'Background', 0.2, -1.0, False, 0.4, 0.4, 0.2, 0.2, '', '', 'Background',
                         0.2, -1.0, False, 0.4, 0.4, 0.2, 0.2, '', '', 'Background', 0.2, -1.0, False, 0.4, 0.4, 0.2,
                         0.2, '', '', 'Background', 0.2, -1.0, False, 0.4, 0.4, 0.2, 0.2, '', '', 'Background', 0.2,
                         -1.0, False, 0.4, 0.4, 0.2, 0.2, '', '', 'Background', 0.2, -1.0, False, 0.4, 0.4, 0.2, 0.2,
                         '', '', 'Background', 0.2, -1.0, True, 1024, 192, True, True, True, True, False, 7, 100,
                         'Constant', 0, 'Constant', 0, 4, False, 'x264', 'blend', 10, 0, 0, False, True, True, True,
                         'intermediate', 'animation', <controlnet.py.UiControlNetUnit object at 0x0000016BA1418A30>,
                         <controlnet.py.UiControlNetUnit object at 0x0000016C30FDB130>, <controlnet.py.UiControlNetUnit
                         object at 0x0000016C307D4130>, <controlnet.py.UiControlNetUnit object at 0x0000016C30FDA860>,
                         False, False, 'Horizontal', '1,1', '0.2', False, False, False, 'Attention', False, '0', '0',
                         '0.4', None, False, 0.75, 1, '<ul>\n<li><code>CFG Scale</code> should be 2 or
                         lower.</li>\n</ul>\n', True, True, '', '', True, 50, True, 1, 0, False, 4, 0.5, 'Linear',
                         'None', '<p style="margin-bottom:0.75em">Recommended settings: Sampling Steps: 80-100,
                         Sampler: Euler a, Denoising strength: 0.8</p>', 128, 8, ['left', 'right', 'up', 'down'], 1,
                         0.05, 128, 4, 0, ['left', 'right', 'up', 'down'], False, False, 'positive', 'comma', 0, False,
                         False, '', '<p style="margin-bottom:0.75em">Will upscale the image by the selected scale
                         factor; use width and height sliders to set tile size</p>', 64, 0, 2, 1, '', [], 0, '', [], 0,
                         '', [], True, False, False, False, 0, False, None, False, None, False, None, False, None,
                         False, 50, False, 4.0, '', 10.0, 'Linear', 3, False, 30.0, True, False, False, 0, 0.0,
                         'Lanczos', 1, True, 0, 0, 0.001, 75, 0.0, False, True, '<p style="margin-bottom:0.75em">Will
                         upscale the image depending on the selected target size type</p>', 512, 0, 8, 32, 64, 0.35,
                         32, 0, True, 0, False, 8, 0, 0, 2048, 2048, 2) kwargs={}
15:55:33-432623 ERROR    gradio call: AssertionError
┌───────────────────────────────────────── Traceback (most recent call last) ─────────────────────────────────────────┐
│ D:\AIModels\Art\StableDiffusion\vladmandic\modules\call_queue.py:33 in f                                            │
│                                                                                                                     │
│    32 │   │   │   try:                                                                                              │
│ >  33 │   │   │   │   res = func(*args, **kwargs)                                                                   │
│    34 │   │   │   │   progress.record_results(id_task, res)                                                         │
│                                                                                                                     │
│ D:\AIModels\Art\StableDiffusion\vladmandic\modules\img2img.py:161 in img2img                                        │
│                                                                                                                     │
│   160 │   │   if processed is None:                                                                                 │
│ > 161 │   │   │   processed = process_images(p)                                                                     │
│   162 │   p.close()                                                                                                 │
│                                                                                                                     │
│                                              ... 23 frames hidden ...                                               │
│                                                                                                                     │
│ D:\AIModels\Art\StableDiffusion\vladmandic\extensions-builtin\sd-webui-controlnet\scripts\hook.py:233 in forward    │
│                                                                                                                     │
│   232 │   │   │   # Handle cond-uncond marker                                                                       │
│ > 233 │   │   │   cond_mark, outer.current_uc_indices, context = unmark_prompt_context(context                      │
│   234 │   │   │   # print(str(cond_mark[:, 0, 0, 0].detach().cpu().numpy().tolist()) + ' - ' +                      │
│                                                                                                                     │
│ D:\AIModels\Art\StableDiffusion\vladmandic\extensions-builtin\sd-webui-controlnet\scripts\hook.py:51 in             │
│ unmark_prompt_context                                                                                               │
│                                                                                                                     │
│    50 def unmark_prompt_context(x):                                                                                 │
│ >  51 │   assert prompt_context_is_marked(x), 'ControlNet Error: Failed to detect whether an i                      │
│    52 │   mark = x[:, 0, :]                                                                                         │
└─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┘
AssertionError: ControlNet Error: Failed to detect whether an instance is cond or uncond!
```

**Comment by crazygonuts** [comment]
this is a recent issue, I was able to use this without problem 2 days ago

**Comment by rushuna86** [comment]
it's commit bccb1a1 from controlnet that breaks it.

**Comment by diego-marinic** [comment]
If you disable controlnet you can run the upscale using noise inversion. Then, enable controlnet tile again and if you don't modify anything else, it will use the cached noise and will work.

**Comment by rushuna86** [comment]
The above works but the actual results are different.

**Comment by wardensc2** [comment]
> it's commit bccb1a1 from controlnet that breaks it.

How do I revert back the controlnet

**Comment by rushuna86** [comment]
using the git checkout command

**Comment by lllyasviel** [comment]
[From ControlNet] this problem is fixed in 1.1.189
We try best to test all other extensions as much as possible and solve conflicting problems

**Comment by rushuna86** [comment]
@lllyasviel thank you for the hard work, can confirm noise inversion working like normal now.

**Comment by wardensc2** [comment]
> [From ControlNet] this problem is fixed in 1.1.189 We try best to test all other extensions as much as possible and solve conflicting problems

Thank you so much for your effort
